lib/resources/aws/aws_s3_bucket.rb

```diff
@@ -89,6 +94,16 @@ def fetch_bucket_policy
     end
   end
 
+  def fetch_bucket_encryption_configuration
+    catch_aws_errors do
+      begin
+        @has_default_encryption_enabled ||= !BackendFactory.create(inspec_runner).get_bucket_encryption(bucket: bucket_name).server_side_encryption_configuration.nil?
+      rescue Aws::S3::Errors::ServerSideEncryptionConfigurationNotFoundError
+        @has_default_encryption_enabled = false
+      end
+    end
+  end
```

Comment:
A few things here. =^)

Line 100 is hard to read for how long it is. Going vertical with those method calls off `BackendFactory` would make it more readable.

We can do a single assignment to `@has_default_encryption_enabled` from whatever you return in the `catch_aws_errors` block.

```ruby
def fetch_bucket_encryption_configuration
  @has_default_encryption_configuration ||= catch_aws_errors do
    begin
      !BackendFactory.create(inspec_runner)
        .get_bucket_encryption(bucket: bucket_name)
        .server_side_encryption_configuration
        .nil?
    rescue Aws::S3::Errors::ServerSideEncryptionConfigurationNotFoundError
      false
    end
  end
end
```